### PR TITLE
CASMINST-6595 - upgrade k8s after iuf deploy product

### PR DESCRIPTION
**Pull request by dzou-hpe**
# Description

# Checklist

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Review by kburns-hpe — APPROVED** [review]

**Review by bklei — APPROVED** [review]

File: operations/iuf/workflows/deploy_product.md [hunk 1]
```diff
@@ -22,7 +22,11 @@ Once this step has completed:
 - New versions of product microservices have been deployed
 - Per-stage product hooks have executed for the `deploy-product` stage
 
-## 2. Next steps
+## 2. Upgrade Kubernetes
+
+1. Follow the steps documented in [Stage 3.6 - Complete Kubernetes upgrade](../../../upgrade/Stage_3.md#stage-36---complete-kubernetes-upgrade).
```
Comment (rustydb) [inline operations/iuf/workflows/deploy_product.md]:
```suggestion
Follow the steps documented in [Stage 3.6 - Complete Kubernetes upgrade](../../../upgrade/Stage_3.md#stage-36---complete-kubernetes-upgrade).
```

Comment (rustydb) [reply operations/iuf/workflows/deploy_product.md]: No need to make it an ordered list of one element.

File: operations/iuf/workflows/deploy_product.md [hunk 2]
```diff
@@ -22,7 +22,11 @@ Once this step has completed:
 - New versions of product microservices have been deployed
 - Per-stage product hooks have executed for the `deploy-product` stage
 
-## 2. Next steps
+## 2. Upgrade Kubernetes
```
Comment (rustydb) [inline operations/iuf/workflows/deploy_product.md:26]: Fix table of contents to align with new header name for `2`.

**Review by rustydb — CHANGES_REQUESTED** [review]
broken toc and minor tweak

**Pull request merged**